src/container/srv_target.c

```diff
@@ -733,6 +734,7 @@ cont_child_free_ref(struct daos_llink *llink)
 	D_DEBUG(DF_DSMS, DF_CONT": freeing\n",
 		DP_CONT(cont->sc_pool->spc_uuid, cont->sc_uuid));
 
+	dtx_cont_deregister(cont);
```

Comment:
I'm not sure why the 'dtx_cont_de/register() are still needs be exported, can't we just call dtx_cont_open() on first cont open and dtx_cont_close() on last cont close.

Reply:
In container level, we cache the container open handle, then even if the application closed the container, there is still one reference on the vos container that will prevent vos container to be closed as to related DRAM resource attached to the vos container cannot be released after application closed the container.

So in this patch, when the application closed the container, it will trigger dtx_cont_close(), that will notify the VOS container to release related DRAM resource. The dtx_cont_deregister() is used to remove the container instance from DTX when the container open handle is removed from the cache.